src/mom5/ocean_param/lateral/ocean_submesoscale.F90

```diff
@@ -2470,10 +2474,11 @@ subroutine compute_flux_x(Time,n,Tracer)
   integer :: ip, kr
   real    :: tensor_13(isd:ied,jsd:jed,0:1)
   real    :: sumz(isd:ied,jsd:jed,0:1)
-
   flux_x = 0.0
 
-  do k=1,nk
+
+  do k=1,min(maxval(kblt(isc-1:iec,jsc:jec)),nk) ! Only need to compute in surface layer but it will not reproduce for some reason
```

Comment:
Is this comment redundant now? It is reproducing right?

Reply:
Correct